### DISCUSSION TITLE
call action with typed params

### DISCUSSION THREAD

**Discussion by SteveSuv** [colyseus/colyseus#723] [Ideas]
I can get state in client:
```ts
this.room.state.players.get(room.sessionId).x
```

why not use action like this:
```ts
this.room.action.startMove({ direction: 1 })
```

in server we can write:
```ts
export class CmdStartMove extends Command<GameRoom, { direction: number }> {
  execute({ client, direction }) {
      const player = this.state.players.get(client.sessionId);
      player.position.x += direction * SPEED
  }
}

export class GameRoomAction extends Action<GameRoom> {
    startMove = CmdStartMove
}

this.setState(new GameRoomState());
this.setAction(new GameRoomAction());
```

then the client not have to use `room.send`, but call the action with typed params.

it's be inspired by [trpc](https://trpc.io/)

**Comment by endel** [comment]
Hi @SteveSuv, thanks for your suggestions. 

A trpc-inspired API would be super nice to have on the "final" Colyseus 1.0 version. I'd love to get there eventually. I think I'll be able to progress towards this only next year, though.
